twitter/searcher.py

```diff
@@ -0,0 +1,183 @@
+    # TODO: include result_type? (mixed, recent, popular) probably want recent once batching is set up
+    # TODO: why wasn't count working
+    return tweepy.Cursor(api.search, q=query, lang="en", since_id=since_id, include_entities=True).items()
+
+
+def convert_tweet(tweet: Dict[str, Any], processed_id_tweets: Set[int]) -> Tuple[List[RawSubmission], Set[int]]:
+    submissions = []
+    # TODO: add "monetizable" logger. like, is somebody profiting off this shit?
+    if not is_tweet_relevant(tweet):
+        return None, processed_id_tweets
+    if "retweeted_status" in tweet:
+        tweet = tweet["retweeted_status"]
+    id_tweet = tweet["id"]
+    if id_tweet in processed_id_tweets:
+        return None, processed_id_tweets
+    processed_id_tweets.add(id_tweet)
+
+    converted_date = twitter_created_at_to_utc(tweet["created_at"])
+    submission_body = ""
+    if tweet["geo"]:
+        submission_body = f"Provided geo: {tweet['geo']}"
+    elif tweet["user"].get("location"):
+        submission_body = f"User Location: {tweet['user']['location']}"
+
+    for media in tweet["extended_entities"]["media"]:
+        if media["type"] != "video":
+            continue
+        print("found video for tweet: ", tweet["text"])
+        media_url = get_media_url_from_variants(media["video_info"]["variants"])
+        if not media_url:
+            continue
+
+        submission = RawSubmission(
+            data_source=DataSource.twitter,
+            id_source=id_tweet,
+            submission_title=tweet["text"],
+            submission_datetime_utc=converted_date,
+            submission_community="",  # TODO: is this ok? I
+            submission_url=media["expanded_url"],
+            submission_media_url=media_url,
+            submission_body=submission_body,
+            id_submitter=tweet["user"]["id"],
+        )
+        submissions.append(submission)
+    return submissions, processed_id_tweets
+
+
+def is_tweet_relevant(tweet: Dict[str, Any]):
+    media = tweet.get("extended_entities", {}).get("media")
+    if media:
+        return True
+    # TODO: do we need a regex like reddit or is that handled by search params
+    return False
+
+
+def twitter_created_at_to_utc(created_at: str):
+    # created at format: "Mon Aug 06 19:28:16 +0000 2018",
+    return datetime.strptime(created_at, "%a %b %d %H:%M:%S %z %Y")
+
+
+def get_media_url_from_variants(variants: List[Dict[str, str]]):
+    for variant in variants:
+        if variant["content_type"] == "video/mp4":
+            # TODO: multiple bitrates are available, does it matter which?
+            return variant["url"]
+    return None
+
+
+# TODO: hook up to s3 bucket
+def log_last_processed_id(last_processed_id: int, last_processed_time_stamp: str):
+    msg = {"last_processed_id": last_processed_id, "last_processed_time_stamp": last_processed_time_stamp}
+    with open(TWITTER_LAST_RUN_FILENAME, "w") as f:
+        f.write(json.dumps(msg))
+    # s3_client = boto_client("s3")
```

Comment:
If It would be nice if these two modes could toggled on and off with configuration